pkg/lib-volume-populator/populator-machinery/controller.go

```diff
@@ -614,6 +647,28 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 
 			// We'll get called again later when the pod exists
 			return nil
+		} else {
+			if pod.Status.PodIP != "" {
+				if _, ok := monitoredPVCs[string(pvc.UID)]; !ok {
+					monitoredPVCs[string(pvc.UID)] = true
+					go func() {
+						c.recorder.Eventf(pod, corev1.EventTypeWarning, reasonPopulatorProgress, "Starting to monitor progress for PVC %s", pvc.Name)
+						for {
+							c.updateProgress(pvc, pod.Status.PodIP, unstructured)
+							pod, err = c.podLister.Pods(populatorNamespace).Get(pod.Name)
+							if err != nil {
+								break
+							}
+							if pod.Status.Phase != corev1.PodRunning {
+								break
+							}
+
+							// TODO make configurable?
+							time.Sleep(5 * time.Second)
```

Comment:
this takes a different approach than the monitoring of virt-v2v as the latter initiates a call in each reconciliation cycle, right? wouldn't it be enough to do that when we reconcile the PVC?

Reply:
Because the populator controller is callback based, it's not reconciled periodically, so without this we'll get only one read...

Reply:
ack